File: examples/python/recon_get.py
import os, json, uuid, time, requests, sys
from datetime import date, timedelta
from dotenv import load_dotenv
sys.path.append(os.path.join(os.path.dirname(__file__), '..', '..'))
from lib.python.signing import sign_request

load_dotenv()

BASE=os.environ["TOKAPAY_BASE_URL"]
CLIENT=os.environ["TOKAPAY_CLIENT_ID"]
with open(os.environ["TOKAPAY_PRIVATE_KEY_PEM_PATH"],"r") as f: PRIV=f.read()
KV=os.getenv("TOKAPAY_KEY_VERSION","1")

def post(path, body):
    body_json=json.dumps(body,separators=(',',':'))
    rid=uuid.uuid4().hex; rtime=str(int(time.time()*1000))
    sig=sign_request("POST", path, CLIENT, rid, rtime, body_json, PRIV, key_version=KV)
    try:
        resp=requests.post(BASE+path, json=body, headers={
            "Client-Id":CLIENT,"Request-Id":rid,"Request-Time":rtime,"Signature":sig,"Content-Type":"application/json"
        }, timeout=15)
        resp.raise_for_status(); return resp.json()
    except requests.exceptions.RequestException as e:
        print(f"ERROR: {e.response.json() if e.response else e}"); return None

if __name__ == "__main__":
    yesterday = date.today() - timedelta(days=1)
    recon_date = yesterday.strftime("%Y-%m-%d")

    response = post("/v1/acquiring/recon/get", {
        "reconDate": recon_date,
        "reconType": "TRANSACTION_DETAIL"
    })
    if response:
        print("SUCCESS:", json.dumps(response, indent=2))

File: examples/node/reconGet.ts
import "dotenv/config";
import fs from "fs";
import axios from "axios";
import crypto from "crypto";
import { signRequest } from "../../lib/node/signing.js";

const BASE = process.env.TOKAPAY_BASE_URL!;
const CLIENT = process.env.TOKAPAY_CLIENT_ID!;
const PRIV = fs.readFileSync(process.env.TOKAPAY_PRIVATE_KEY_PEM_PATH!, "utf8");

async function main() {
  const path = "/v1/acquiring/recon/get";
  const yesterday = new Date();
  yesterday.setDate(yesterday.getDate() - 1);
  const reconDate = yesterday.toISOString().split('T')[0]; // YYYY-MM-DD

  const body = {
    reconDate: reconDate,
    reconType: "TRANSACTION_DETAIL"
  };
  const bodyStr = JSON.stringify(body);
  const requestId = crypto.randomUUID().replace(/-/g, "");
  const requestTime = Date.now().toString();

  const sig = signRequest({
    method: "POST",
    path,
    clientId: CLIENT,
    requestId,
    requestTimeMs: requestTime,
    bodyJson: bodyStr,
    privateKeyPem: PRIV,
    keyVersion: process.env.TOKAPAY_KEY_VERSION || "1"
  });

  try {
    const res = await axios.post(BASE + path, body, {
      headers: {
        "Content-Type": "application/json",
        "Client-Id": CLIENT,
        "Request-Id": requestId,
        "Request-Time": requestTime,
        "Signature": sig
      }
    });
    console.log("SUCCESS:", res.data);
  } catch (e: any) {
    console.error("ERROR:", e.response?.data || e.message);
  }
}

main();
